extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -761,6 +767,69 @@ export class SchemaCompareResult {
 		});
 	}
 
+	private createSaveScmpButton(view: azdata.ModelView): void {
+		this.saveScmpButton = view.modelBuilder.button().withProperties({
+			label: localize('schemaCompare.saveScmpButton', 'Save .scmp file'),
+			iconPath: {
+				light: path.join(__dirname, 'media', 'save-scmp.svg'),
+				dark: path.join(__dirname, 'media', 'save-scmp-inverse.svg')
+			},
+			title: localize('schemaCompare.saveScmpButtonTitle', 'Save source and target, options, and excluded elements'),
+			enabled: false
+		}).component();
+
+		this.saveScmpButton.onDidClick(async (click) => {
+			const rootPath = vscode.workspace.rootPath ? vscode.workspace.rootPath : os.homedir();
```

Comment:
Does this work even if you don't have a workspace open?

Reply:
yup it will use the folder opened in the file explorer